openmp/runtime/test/worksharing/for/collapse_test.inc

```diff
@@ -106,12 +106,14 @@ int test() {
   unsigned scalarCount = 0;
   unsigned uselessThreadsOpenMP = 0;
   unsigned usefulThreadsOpenMP = 0;
-  unsigned chunkSizesOpenmp[MAX_THREADS] = {0};
 
-  unsigned num_threads = omp_get_max_threads();
+  // Use half of the available threads/logical processor.
+  unsigned num_threads = omp_get_max_threads() / 2;
   if (num_threads > MAX_THREADS)
     num_threads = MAX_THREADS;
-  omp_set_num_threads(num_threads);
+
+  unsigned *chunkSizesOpenmp = malloc(sizeof(unsigned) * num_threads);
```

Comment:
If you malloc, you should also free. I don't see anything wrong with the previous stack allocation approach.

To reduce the impact of cache line sharing, it would help to use long instead of int.

Reply:
> If you malloc, you should also free. I don't see anything wrong with the previous stack allocation approach.
> 
Good catch, will fix it. Thanks!
> To reduce the impact of cache line sharing, it would help to use long instead of int.

Did you mean to change `unsigned *chunkSizesOpenmp` to `unsigned long *chunkSizeOpenmp`?

Reply:
>>  To reduce the impact of cache line sharing, it would help to use long instead of int.
>
> Did you mean to change unsigned *chunkSizesOpenmp to unsigned long *chunkSizeOpenmp?

Hi @jprotze, I've addressed comments except the one quoted above. It would be appreciated if you can clarify at your convenience.

Reply:
Yes, that's the change I proposed.

Reply:
Thanks, @jprotze! I've changed the type of `chunkSizeOpenmp` to `unsigned long` and refreshed the patch.